pkg/didcomm/transport/http/inbound.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package http
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+// inboundCommHTTPOpts struct for path and handler represented
+// as an inbound http option
+type inboundCommHTTPOpts struct {
+	Path        string
+	HandlerFunc http.HandlerFunc
+}
+
+// InboundHTTPOpt is an inbout HTTP transport option
+type InboundHTTPOpt func(opts *inboundCommHTTPOpts)
+
+// DIDCommHandler struct to pass details for request handlers
+type DIDCommHandler struct {
+	routersOpts []*inboundCommHTTPOpts
+}
+
+// WithInboundRoute option is for creating an Inbount HTTP transport using an http.Handler
+func WithInboundRoute(inboundPath string, handlerFunc http.HandlerFunc) InboundHTTPOpt {
+	return func(opts *inboundCommHTTPOpts) {
+		opts.Path = inboundPath
+		opts.HandlerFunc = handlerFunc
+	}
+}
+
+// NewInboundHandler will create a new handler to enforce Did-Comm HTTP transport specs
+// then routes processing to the passed in handler argument
+func NewInboundHandler(opts ...InboundHTTPOpt) (http.Handler, error) {
+	var routersOpts []*inboundCommHTTPOpts
+	// Apply options
+	for _, opt := range opts {
+		routerOpts := &inboundCommHTTPOpts{}
+		if opt == nil {
+			return nil, errors.New("router option is nil")
+		}
+		opt(routerOpts)
+		routersOpts = append(routersOpts, routerOpts)
+	}
+
+	commHandler := &DIDCommHandler{routersOpts: routersOpts}
+	err := validateHandler(commHandler)
+	if err != nil {
+		// TODO add log message here
+		return nil, errors.Wrapf(err, "Failed to create NewInboundHandler")
+	}
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
@Baha-sk this design would force the user to bind this handler to the root path `/` or some other path that is parent of the specific handler paths. A much simpler design would be to simply provide handlers and let the user take care of binding them to the paths. It also avoids the potential DoS due to the `return` on line 65 - this inbound connection should not be dropped because a request was made to an unregistered path - that is outside the user's control.

Note: for the demo, we definitely need to serve non-DIDComm paths

Reply:
since DidComm should only have 1 URL path for inbound communication, the code was refactored to allow for only 1 path and its corresponding http handler.

as discussed offline @llorllale, did exchange routing should be done at a higher level where the payload envelope is unmarshalled as it has the routing info required by DidComm.

Reply:
update, it seems no URL path is required as agents will communicate via 1 endpoint (regardless of the path). As mentioned above, DID Exchange flow routing will be done at a higher level. 

So all that's required here is a simple msgHandler function that will manage the payload (envelope) handling.